packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -719,6 +721,24 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
     });
   }
 
+  /// Simulates sending a physical key down event through the system channel.
+  ///
+  /// This only simulates key presses coming from a physical keyboard,
+  /// not from a soft keyboard.
+  Future<void> sendKeyDownEvent(LogicalKeyboardKey key, { String platform = 'android'}) {
+    // Internally wrapped in async guard.
+    return simulateKeyDownEvent(key, platform: platform);
+  }
+
+  /// Simulates sending a physical key up event through the system channel.
+  ///
+  /// This only simulates key presses coming from a physical keyboard,
```

Comment:
nit: "This only simulates key lifts coming from a physical keyboard,"

Reply:
I re-worded it as "key up events", because I haven't really heard "lifts" used this way.